sdk/logs/src/main/java/io/opentelemetry/sdk/logs/SdkLogEmitterProvider.java

```diff
@@ -33,8 +34,12 @@ public static SdkLogEmitterProviderBuilder builder() {
     return new SdkLogEmitterProviderBuilder();
   }
 
-  SdkLogEmitterProvider(Resource resource, List<LogProcessor> processors, Clock clock) {
-    this.sharedState = new LogEmitterSharedState(resource, processors, clock);
+  SdkLogEmitterProvider(
+      Resource resource,
+      Supplier<LogLimits> logLimitsSupplier,
+      List<LogProcessor> processors,
+      Clock clock) {
+    this.sharedState = new LogEmitterSharedState(resource, logLimitsSupplier, processors, clock);
```

Comment:
I think creating new instances in the constructor is a bit of a code smell (DI violation). If it's unreasonable to make the callers do this creation to pass in the real dependency (`LogEmitterSharedState`) then maybe a factory method is in order?

Reply:
The only caller of this constructor is `SdkLogEmitterProviderBuilder#build()`. In this case requiring the caller to construct the `LogEmitterSharedState` shifts the builder from being a simple data carrier for configuration options to it knowing about the internals of the `SdkLogEmitterProvider`.